frameworks/JavaScript/express/graphql-mysql-app.js

```diff
@@ -0,0 +1,295 @@
+});
+
+app.get('/fortunes', async (req, res) => {
+
+    const graphql = await graphqlHTTP((req, res, graphQLParams) => {
+
+        graphQLParams.query = "{ getAllFortunes { id, message } }"
+        graphQLParams.operationName = null;
+        graphQLParams.variables = {};
+
+        return graphqlOpts();
+    });
+
+    retrieveAndFormatAllFortunes(res);
+    
+    await graphql(req, res);
+});
+
+app.get('/updates', async (req, res) => {
+
+    const totalNumOfQueries = ensureQueryIsAnInt(req.query.queries);
+
+    const graphql = await graphqlHTTP((req, res, graphQLParams) => {
+
+        graphQLParams.query = `{ getRandomAndUpdate(total: ${totalNumOfQueries}) { id, randomNumber } }`
+        graphQLParams.operationName = null;
+        graphQLParams.variables = {};
+
+        return graphqlOpts();
+    });
+
+    retrieveUpdateAndFormatWorlds(res);
+
+    await graphql(req, res);
+});
+
+
+app.get('/plaintext', (req, res) => {
+
+    const responseTxt = "Hello, World!";
+
+    res.setHeader('Content-Length', responseTxt.length);
+    res.setHeader('Server', 'Express-GraphQL-MySQL');
+    res.contentType('text/plain');
+    res.status(200);
+
+    res.send(responseTxt);
+});
+
+// Helper Functions
+
+const ensureQueryIsAnInt = (queryString) => {
+
+    if(queryString === undefined) return 1;
+
+    const possibleInt = parseInt(queryString);
+    if(!possibleInt) return 1;
+
+    return possibleInt;
+};
+
+const graphqlOpts = (params) => {
+    return {
+        schema,
+        graphiql: false,
+        context: params || {}
+    }
+};
+
+const retrieveUpdateAndFormatWorlds = res => {
+
+    res.real_end = res.end;
+
+    res.end = data => {
+        
+        var toRet;
+        const json = JSON.parse(data.toString('utf8'));
+
+        if(json.data.getRandomAndUpdate) {
+            toRet = json.data.getRandomAndUpdate;
+        } else {
+            toRet = [];
+        }
+
+        const jsonToRet = JSON.stringify(toRet);
+
+        setResponseHeaders(res, jsonToRet.length);
+
+        res.real_end(jsonToRet);
+    }
+};
+
+const retrieveAndFormatAllFortunes = res => {
+
+    res.real_end = res.end;
+
+    res.end = async (data) => {
+        var toRet;
+        const json = JSON.parse(data.toString('utf8'));
+
+        if(json.data.getAllFortunes) {
+            toRet = json.data.getAllFortunes;
+        } else {
+            toRet = [];
+        }
+
+        const newFortune = { "id": 0, "message": "Additional fortune added at request time." };
+        toRet.push(newFortune);
+        toRet.sort((a, b) => (a.message < b.message) ? -1 : 1);
+
+        const htmlToRet = await spoofHTML(toRet);
+
+        res.contentType('html');
+        res.setHeader('Server', 'GraphQL-MySQL');
+        res.setHeader('Content-Length', htmlToRet.length + 32);
+        res.status(200);
+
+        res.real_end(htmlToRet);
+    }
+};
+
+const spoofHTML = arr => {
+
+    return new Promise((resolve, reject) => {
+
+        var count = 0;
```

Comment:
Super JS nit for me: we have `let`, `const`, and `var` all being used here.. can we replace all the `var` declarations with `let`